flink-connector-kudu/src/main/java/org/apache/flink/connector/kudu/source/enumerator/KuduSplitGenerator.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.connector.kudu.source.enumerator;
+
+import org.apache.flink.connector.kudu.connector.KuduTableInfo;
+import org.apache.flink.connector.kudu.connector.reader.KuduReaderConfig;
+import org.apache.flink.connector.kudu.source.split.KuduSourceSplit;
+
+import org.apache.kudu.client.AsyncKuduScanner;
+import org.apache.kudu.client.KuduClient;
+import org.apache.kudu.client.KuduScanToken;
+import org.apache.kudu.client.KuduTable;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * The class responsible for producing scan tokens for given timestamps and returning them in the
+ * form of {@link KuduSourceSplit}.
+ */
+public class KuduSplitGenerator {
+    private final KuduTableInfo tableInfo;
+    private final KuduClient kuduClient;
+
+    public KuduSplitGenerator(KuduReaderConfig readerConfig, KuduTableInfo tableInfo) {
+        this.tableInfo = tableInfo;
+        this.kuduClient = new KuduClient.KuduClientBuilder(readerConfig.getMasters()).build();
+    }
+
+    public List<KuduSourceSplit> generateFullScanSplits(long snapshotTimestamp) {
+        try {
+            KuduTable table = kuduClient.openTable(tableInfo.getName());
+            List<KuduScanToken> tokens =
+                    kuduClient
+                            .newScanTokenBuilder(table)
+                            .snapshotTimestampRaw(snapshotTimestamp)
+                            .readMode(AsyncKuduScanner.ReadMode.READ_AT_SNAPSHOT)
+                            .build();
+
+            return serializeTokens(tokens);
+        } catch (Exception e) {
+            throw new RuntimeException("Error during full snapshot scan", e);
```

Comment:
is there a way to extract more information out of this exception? In the related tests every scenario tests for the general error message "Error during full snapshot scan", would it be possible to change this to be a bit more user-friendly?